crates/sui-core/src/authority.rs

```diff
@@ -3003,7 +3012,10 @@ impl AuthorityState {
             let new_object = Object::new_system_package(
                 bytes
                     .iter()
-                    .map(|m| CompiledModule::deserialize(m).unwrap())
+                    .map(|m| {
```

Comment:
I've made some pretty extensive changes to how framework upgrades will work in #9366 -- could you base your changes off of those?